modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/BindingListener.java

```diff
@@ -18,31 +18,66 @@
 
 package org.wso2.andes.kernel;
 
+import org.apache.log4j.Logger;
+import org.wso2.andes.server.ClusterResourceHolder;
+import org.wso2.andes.server.cluster.coordination.ClusterEventNotifier;
+import org.wso2.andes.server.cluster.coordination.ClusterNotification;
+
 /**
- * Listener listening for binding changes local and cluster
+ * BindingListener listens and handles binding changes that have occurred locally and cluster-wide.
  */
-public interface BindingListener {
+public class BindingListener {
+
+    Logger log = Logger.getLogger(BindingListener.class);
 
-    public static enum BindingEvent {
+    public enum BindingEvent {
         ADDED,
         DELETED
     }
 
     /**
-     * handle when an binding has changed in cluster
+     * Local event handler is used to notify local binding changes to the cluster.
+     */
+    private ClusterEventNotifier clusterEventNotifier;
+
+    public BindingListener(ClusterEventNotifier handler) {
+        clusterEventNotifier = handler;
+    }
+
+    /**
+     * Handle when a binding has changed in cluster.
      *
      * @param binding    binding changed
      * @param changeType the change
      * @throws AndesException
      */
-    public void handleClusterBindingsChanged(AndesBinding binding, BindingEvent changeType) throws AndesException;
+    public void handleClusterBindingsChanged(AndesBinding binding, BindingEvent changeType) throws AndesException {
+        if (log.isDebugEnabled()) {
+            log.debug("Cluster event received: " + binding.encodeAsString());
+        }
+        switch (changeType) {
+            case ADDED:
+                //create a binding
+                ClusterResourceHolder.getInstance().getVirtualHostConfigSynchronizer().clusterBindingAdded(binding);
+                break;
```

Comment:
May be we can use a field for ClusterResourceHolder.getInstance().getVirtualHostConfigSynchronizer() since it does not change in run time.